strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/configuration/ConfigurationRepository.java

```diff
@@ -57,12 +69,30 @@ public synchronized void init()
     {
         logger.debug("ConfigurationRepository.init()");
 
-        getDatabase().getEntityManager().registerEntityClass(BinaryConfiguration.class, true);
+        oEntityManager.registerEntityClass(BinaryConfiguration.class, true);
 
-        if (!schemaExists() || getConfiguration() == null)
+        transactionTemplate.execute(new TransactionCallback<Object>()
         {
-            createSettings();
-        }
+
+            @Override
+            public Object doInTransaction(TransactionStatus status)
+            {
+                if (!schemaExists() || getConfiguration() == null)
+                {
+                    try
+                    {
+                        createSettings();
+                    }
+                    catch (IOException e)
+                    {
+                        throw new RuntimeException(e);
```

Comment:
@sbespalov : I agree. I don't think a `RuntimeException` would be the best in this case...